docs/newbs_git_workflow.md

```diff
@@ -0,0 +1,83 @@
+# Building QMK with GitHub Actions
+A workflow can be setup for GitHub Actions to automatically build QMK firmware. To use this feature, a GitHub account must be setup [using a personal fork of QMK](getting_started_github.md).
+
+# Setting up the workflow
+The workflow should be setup in a personal or development branch, not the `master`. It should be placed in the same user keymap development branch. See [Git best practices for working with QMK](newbs_git_best_practices.md) for details.
+
+Create a `build.yml` in the folder location `~/qmk_firmware/.github/workflows`, with the following content:
+
+```yml
+name: Build firmware for keyboards
+on: [push, workflow_dispatch]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container: qmkfm/qmk_cli
+    strategy:
+      fail-fast: false
+      matrix:
+        keyboard:
+          - clueboard/66/rev4
+        user:
+          - ${{ github.actor }}
+
+    steps:
+
+    - name: Checkout QMK
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 1
+        persist-credentials: false
+        submodules: true
+
+    - name: Build firmware
+      run: qmk compile -kb ${{ matrix.keyboard }} -km ${{ matrix.user }}
+
+    - name: Archive firmware
+      uses: actions/upload-artifact@v2
+      with:
+        name: "${{ matrix.keyboard }}_${{ matrix.user }}_firmware"
```

Comment:
This will error out if it's a subfolder. 

Eg, when compiling for `40percentclub/nano`: 
> Error: Artifact name is not valid: 40percentclub/nano_drashna_firmware. Contains the following character:  Forward slash /

Reply:
The output will group all firmware by user zip file.